src/org/thoughtcrime/securesms/connect/AccountManager.java

```diff
@@ -0,0 +1,293 @@
+        } catch(Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    // ui
+
+    public void handleSwitchAccount(Activity activity) {
+        ArrayList<AccountManager.Account> accounts = getAccounts(activity);
+
+        // build menu
+        int presel = 0;
+        ArrayList<String> menu = new ArrayList<>();
+        for (int i = 0; i < accounts.size(); i++) {
+            AccountManager.Account account = accounts.get(i);
+            if (account.isCurrent()) {
+                presel = i;
+            }
+            menu.add(account.getDescr(activity));
+        }
+
+        int addAccount = menu.size();
+        menu.add(activity.getString(R.string.add_account));
+
+        // show dialog
+        AlertDialog.Builder builder = new AlertDialog.Builder(activity)
+                .setTitle(R.string.switch_account)
+                .setNegativeButton(R.string.cancel, null)
+                .setSingleChoiceItems(menu.toArray(new String[menu.size()]), presel, (dialog, which) -> {
+                    if (which==addAccount) {
+                        beginAccountCreation(activity);
+                        activity.finishAffinity();
+                        activity.startActivity(new Intent(activity, WelcomeActivity.class));
+                    } else { // switch account
+                        switchAccount(activity, accounts.get(which));
+                        activity.finishAffinity();
+                        activity.startActivity(new Intent(activity.getApplicationContext(), ConversationListActivity.class));
+                    }
+                });
+        if (accounts.size() > 1) {
+            builder.setNeutralButton(R.string.delete_account, (dialog, which) -> {
+                handleDeleteAccount(activity);
+            });
+        }
+        builder.show();
+    }
+
+    private void handleDeleteAccount(Activity activity) {
+        ArrayList<AccountManager.Account> accounts = getAccounts(activity);
+
+        ArrayList<String> menu = new ArrayList<>();
+        for (AccountManager.Account account : accounts) {
+            menu.add(account.getDescr(activity));
+        }
+        int[] selection = {-1};
+        new AlertDialog.Builder(activity)
+                .setTitle(R.string.delete_account)
+                .setSingleChoiceItems(menu.toArray(new String[menu.size()]), -1, (dialog, which) -> selection[0] = which)
+                .setNegativeButton(R.string.cancel, (dialog, which) -> handleSwitchAccount(activity))
+                .setPositiveButton(R.string.ok, (dialog, which) -> {
+                    if (selection[0] >= 0 && selection[0] < accounts.size()) {
+                        AccountManager.Account account = accounts.get(selection[0]);
+                        if (account.isCurrent()) {
+                            new AlertDialog.Builder(activity)
+                                    .setMessage("To delete the currently active account, switch to another account first.")
```

Comment:
This needs to be translated at some point!

Reply:
yip, however, it is unclear if the ui stays this way.